components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java

```diff
@@ -1831,40 +2000,37 @@ String[] filterScopeValidators(OAuthConsumerAppDTO application) throws IdentityO
     /**
      * Get the IdToken Encryption Method registered by the user and filter the allowed one.
      *
-     * @param application Application user have registered
+     * @param encryptionMethod Encryption method sent in the registration request.
      * @return idTokenEncryptionMethod
      * @throws IdentityOAuthAdminException Identity OAuthAdmin exception.
      */
-    private String filterIdTokenEncryptionMethod(OAuthConsumerAppDTO application) throws IdentityOAuthAdminException {
+    private String filterEncryptionMethod(String encryptionMethod, String configName)
+            throws IdentityOAuthAdminException {
 
-        List<String> supportedIdTokenEncryptionMethods = OAuthServerConfiguration.getInstance()
-                .getSupportedIdTokenEncryptionMethods();
-        String idTokenEncryptionMethod = application.getIdTokenEncryptionMethod();
-        if (!supportedIdTokenEncryptionMethods.contains(idTokenEncryptionMethod)) {
-            String msg = String.format("'%s' IdToken Encryption Method is not allowed.", idTokenEncryptionMethod);
+        List<String> supportedIdTokenEncryptionMethods = IdentityUtil.getPropertyAsList(configName);
+        if (!supportedIdTokenEncryptionMethods.contains(encryptionMethod)) {
+            String msg = String.format("'%s' Encryption Method is not allowed.", encryptionMethod);
             throw handleClientError(INVALID_REQUEST, msg);
         }
-        return idTokenEncryptionMethod;
+        return encryptionMethod;
     }
 
     /**
      * Get the IdToken Encryption Algorithm registered by the user and filter the allowed one.
      *
-     * @param application Application user have registered
+     * @param algorithm algorithm sent in the registration request.
      * @return idTokenEncryptionAlgorithm
      * @throws IdentityOAuthAdminException Identity OAuthAdmin exception.
      */
-    private String filterIdTokenEncryptionAlgorithm(OAuthConsumerAppDTO application)
+    private String filterEncryptionAlgorithms(String algorithm, String configName)
             throws IdentityOAuthAdminException {
 
-        List<String> supportedIdTokenEncryptionAlgorithms = OAuthServerConfiguration.getInstance()
-                .getSupportedIdTokenEncryptionAlgorithm();
-        String idTokenEncryptionAlgorithm = application.getIdTokenEncryptionAlgorithm();
-        if (!supportedIdTokenEncryptionAlgorithms.contains(idTokenEncryptionAlgorithm)) {
-            String msg = String.format("'%s' IdToken Encryption Method is not allowed.", idTokenEncryptionAlgorithm);
+        List<String> supportedIdTokenEncryptionAlgorithms = IdentityUtil.getPropertyAsList(configName);
```

Comment:
Lets change the variable name `supportedIdTokenEncryptionAlgorithms `

Reply:
changed in https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2178/commits/f6b06f0d8192508c55bf5ed2309a3732f2a16680